search-engines.lisp

```diff
@@ -622,6 +622,243 @@ REQUEST-ARGS is a list of args to pass to request function."
+         (parsed-hex-int-string (format nil "~X" parsed-dec-int)))
+    (if (eql (length parsed-hex-int-string) 160) ; The settings string should be 160 characters long.
+        parsed-hex-int-string
+        (progn (log:warn "The value specified for SETTINGS-STRING is not valid.
+ Defaulting to empty value")
+         ""))))
+
+(defun make-startpage-completion (&key request-args)
+  "Helper that generates Startpage search completion functions.
+REQUEST-ARGS is a list of args to pass to request function."
+  (make-search-completion-function
+   :base-url "https://startpage.com/suggestions?q=~a&segment=startpage.udog"
+   :processing-function
+   #'(lambda (results)
+       (mapcar #'cdar
+               (cddadr (json:decode-json-from-string results))))
+   :request-args request-args))
+
+(define-search-engine startpage
+    (:shortcut "startpage"
+     :fallback-url (quri:uri "https://startpage.com/")
+     :base-search-url "https://startpage.com/do/search?query=~a"
+     :completion-function (make-startpage-completion)
+     :documentation "Startpage `nyxt:search-engine' which can configure the settings accessible from the search page.
+In order to specify settings from Startpage's \"Settings\" page, set `settings-string' to the hexadecimal number situated
+after \"prfe=\" in the URL displayed in the \"Save without cookie\" section.")
+  (object "cat" ((:web "web")
+                 (:images "pics")
+                 (:videos "video")
+                 (:news "news")))
+  (language-ui "lui" ((:dansk "dansk")
+                      (:deutsch "deutsch")
+                      (:english "english")
+                      (:espanol "espanol")
+                      (:francais "francais")
+                      (:nederlands "nederlands")
+                      (:norsk "norsk")
+                      (:polski "polski")
+                      (:portugues "portugues")
+                      (:svenska "svenska")))
+    (language-results "language" ((:afrikaans "afrikaans")
+                                  (:albanian "albanian")
+                                  (:amharic "amharic")
+                                  (:arabic "arabic")
+                                  (:azerbaijani "azerbaijani")
+                                  (:basque "basque")
+                                  (:belarusian "belarusian")
+                                  (:bengali "bengali")
+                                  (:bihari "bihari")
+                                  (:bosnian "bosnian")
+                                  (:bulgarian "bulgarian")
+                                  (:catalan "catalan")
+                                  (:croatian "croatian")
+                                  (:czech "czech")
+                                  (:dansk "dansk")
+                                  (:deutsch "deutsch")
+                                  (:english "english")
+                                  (:english_uk "english_uk")
```

Comment:
Lisp uses `kebab-case`, so search engine keywords should preferable be kebab-cased too. Even though Startpage uses "english_uk", Lisp side should better have `:english-uk`.

Reply:
Done!